Framework/DataHandling/src/SaveNexusProcessed.cpp

```diff
@@ -262,13 +261,18 @@ void SaveNexusProcessed::doExec(
       else
         workspaceTypeGroupName = "workspace";
 
-      nexusFile->writeNexusProcessedData2D(matrixWorkspace, uniformSpectra,
-                                           spec, workspaceTypeGroupName.c_str(),
-                                           true);
+      nexusFile->writeNexusProcessedData2D(
+          matrixWorkspace, uniformSpectra, indices,
+          workspaceTypeGroupName.c_str(), true);
     }
 
     cppFile.openGroup("instrument", "NXinstrument");
-    saveSpectraMapNexus(*matrixWorkspace, &cppFile, spec, ::NeXus::LZW);
+    cppFile.makeGroup("detector", "NXdetector", true);
+    cppFile.putAttr("version", 1);
+    saveSpectraDetectorMapNexus(*matrixWorkspace, &cppFile, indices,
+                                ::NeXus::LZW);
+    saveSpectrumNumbersNexus(*matrixWorkspace, &cppFile, indices, ::NeXus::LZW);
+    cppFile.closeGroup();
```

Comment:
Do we need to call this function twice?

Reply:
Yes, on line 269 we `openGroup` and immediately `makeGroup`. Both groups have to be closed before returning from this function.